gateway/sds_gateway/api_methods/tasks.py

```diff
@@ -1019,10 +1080,16 @@ def send_item_files_email(  # noqa: C901, PLR0912
         error_message = f"Error processing {item_type_enum} download: {e!s}"
         if user is not None and item is not None:
             _send_item_download_error_email(user, item, item_type_enum, error_message)
+        # Mark temp_zip as failed if it exists
+        if "temp_zip" in locals():
+            temp_zip.mark_failed()
         result = _create_error_response("error", error_message, item_uuid)
 
     except (SoftTimeLimitExceeded, TimeoutError) as e:
         result = _handle_timeout_exception(user, item, item_type_enum, item_uuid, e)
+        # Mark temp_zip as failed if it exists
+        if "temp_zip" in locals():
+            temp_zip.mark_failed()
 
     finally:
```

Comment:
we should probably also call `mark_failed` for any uncaught exceptions

```py
except Exception:
    if temp_zip:
        temp_zip.mark_failed()
    raise
```

also, to avoid the `"temp_zip" in locals()` checks and related linter warnings, you can just declare `temp_zip = None` before the try block.